homeassistant/components/modbus/sensor.py

```diff
@@ -243,13 +255,16 @@ def update(self):
             registers.reverse()
 
         byte_string = b"".join([x.to_bytes(2, byteorder="big") for x in registers])
-        val = struct.unpack(self._structure, byte_string)[0]
-        val = self._scale * val + self._offset
-        if isinstance(val, int):
-            self._value = str(val)
-            if self._precision > 0:
-                self._value += "." + "0" * self._precision
+        if self._data_type != DATA_TYPE_STRING:
```

Comment:
This seems complicated especially since “int” is the most common form.

Please keep the existing code and add an else for DATA_TYPE_STRING.

Reply:
Hi @janiversen - Thanks for your prompt review. Is there anything you would like me to action at the moment?

Reply:
We can do the change of isinstance() but it would be real nice to have test cases for the string type. See core/tests/component/modbus

Reply:
Sounds great to me. I am new to python, HA and GitHub, so I might make some mistakes along the way. I'll read up about test cases and give it a go.

Reply:
I've added a commit for a test of the new string data_type